src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -1450,4 +1450,25 @@ namespace winrt::TerminalApp::implementation
         return _root->IsQuakeWindow();
     }
 
+    bool AppLogic::GetMinimizeToTray()
+    {
+        if (!_loadedInitialSettings)
+        {
+            // Load settings if we haven't already
+            LoadSettings();
+        }
+
+        return _settings.GlobalSettings().MinimizeToTray();

```

Comment:
Okay, this is gonna sound crazy, but hear me out - 

can we remove all of the other `#ifdef`s in this PR, and replace them with just one here?
```suggestion
#if TIL_FEATURE_TRAYICON_ENABLED 
        return _settings.GlobalSettings().MinimizeToTray();
#else 
        return false;
#endif
```

(here and in `AppLogic::GetAlwaysShowTrayIcon`)?

The code is a little confusing to read with them scattered throughout. Seems like it'd be easier to have all the code there and ready to go, just permanently disabled when the feature is off.

Reply:
changed em! essentially makes the two settings false if the feature is disabled, got rid of some other places that used the flag and tried to localize it to all the entrypoints to the tray icon creation/deletion.